### PR TITLE
Fix #10177 - Mac OS deployment packages with a target of 10.15 no longer produced 

### LINKED ISSUE #10177: Mac OS deployment packages with a target of 10.15 no longer produced
Issue overview
--------------

Since macos-10.15 runner was removed from GHA, the v23.2.0-IOFreeze is the first release that doesn't have the 10.15 macos x86_64 packages.

Either we should make it clear that we no longer wish to support this (and check downstream users if that is fine), or we need to adjust the CI (my preference, assuming building on macos-11 with a deployment target of 10.15 works fine).

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): macOS x86_64
 - Version of EnergyPlus (if using an intermediate build, include SHA): v23.2.0-IOFreeze

cf https://github.com/NREL/EnergyPlus/actions/runs/5979574269 

image

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #10177 
 
Test run was here: https://github.com/NREL/EnergyPlus/actions/runs/5999567942
Grabbed the 10.15 binaries and reuploaded them to the v23.2.0-IOFreeze release

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [x] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: .github/workflows/mac_release.yml [hunk 1]
```diff
@@ -20,18 +20,17 @@ jobs:
...
+        - macos_dev_target: 10.15
+          os: macos-11
           allow_failure: false
-          macos_dev_target: 11.6
-        - os: macos-12
+        - macos_dev_target: 11.6
+          os: macos-11
+          allow_failure: false
+        - macos_dev_target: 12.1
+          os: macos-12
```
Comment (jmarrec) [inline .github/workflows/mac_release.yml:32]: run the MACOSX_DEPLOYMENT_TARGET=10.15 onto a macos-11 runner.

File: cmake/Install.cmake [hunk 2]
```diff
@@ -34,25 +34,13 @@ if(UNIX)
...
-  # TODO: the "FORCE" is temporary to avoid people having an existing build directory miss the fact that the recommended method changed
-  # TODO: remove after next release
-  if(UNIX AND NOT APPLE)
-    if(NOT CPACK_BINARY_IFW)
-      set(CPACK_BINARY_STGZ OFF CACHE BOOL "This was the legacy method on Linux, superseded by IFW" FORCE)
-      set(CPACK_BINARY_IFW ON CACHE BOOL "Enable to build IFW package, which is the recommend method" FORCE)
-      message("Switching from STGZ to IFW as the supported generator has changed on Linux")
-    endif()
-  endif()
-  # END TODO
```
Comment (jmarrec) [inline cmake/Install.cmake:54]: Long overdue remove

File: .github/workflows/mac_release.yml (hunk 1 above)
Comment (Myoldmopar) [reply .github/workflows/mac_release.yml:32]: Do we have to define the `macos_dev_target` variable first and then also put it inside the `matrix::include` scope too?

File: cmake/Install.cmake [hunk 3]
```diff
@@ -9,7 +9,7 @@
 #elseif(APPLE)
 #set(CPACK_GENERATOR "IFW;TGZ")
 #elseif(UNIX)
-#set(CPACK_GENERATOR "STGZ;TGZ")
+#set(CPACK_GENERATOR "IFW;STGZ;TGZ")
```
Comment (Myoldmopar) [inline cmake/Install.cmake:12]: This is fine, also if these comments aren't needed, we could remove them entirely.

File: cmake/Install.cmake (hunk 2 above)
Comment (Myoldmopar) [reply cmake/Install.cmake:54]: Agreed, I remember when you added this temporary fix.

**Review by Myoldmopar — APPROVED** [review]
This looks functionally ready to go.  I had two cleanup/non-functional comments, but they can be addressed later.  @jmarrec let me know if you want me to just send my comments to /dev/null or if you have any thoughts on them.  This is ready for merge, but I'll hold for a bit to open for comments.

**Comment by Myoldmopar** [comment]
OK, no comments, so I'm merging.  Thanks @jmarrec

File: .github/workflows/mac_release.yml (hunk 1 above)
Comment (jmarrec) [reply .github/workflows/mac_release.yml:32]: Not sure I follow the question. The `macos_dev_target` **is** the matrix, and the include bit just adds other info for each.

This is functionally equivalent:

```
matrix:
  name: [macOS-10.15, macOS-11]
  include:
     - name: macOS-10.15
       os: macos-11
       macos_dev_tatget: 10.15
     - name: macOS-11
       os: macos-11
       macos_dev_tatget: 11.6
```

Comment (Myoldmopar) [reply .github/workflows/mac_release.yml:32]: I'm just saying I don't think you need to define the names first and then repeat them.  You can just list them out directly in the include block.  

```
matrix: 
  include: 
    - name: macOS-10.15 
      os: macos-11
      macos_dev_tatget: 10.15
    - name: macOS-11 
      os: macos-11 
      macos_dev_tatget: 11.6
```

Comment (jmarrec) [reply .github/workflows/mac_release.yml:32]: I didn't know that, thanks. I kinda like having the list(s) up top, makes it easier to see what's going on / how many runs this ends up producing.

**Pull request merged**